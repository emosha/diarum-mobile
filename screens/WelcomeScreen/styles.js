import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  imageBackground: {
    width: '100%',
    height: '100%',
  },
  pagination: {
    backgroundColor: 'transparent',
    position: 'absolute',
    width: '100%',
    bottom: '-5%',
  },
  buttonStyle: {
    backgroundColor: '#F5BDBA',
    borderRadius: 0,
    padding: 15,
    paddingLeft: 40,
    paddingRight: 40,
  },
  buttonContainerStyle: {
    marginTop: 35,
  },
  buttonTitleStyle: {
    textAlign: 'center',
    fontSize: 25,
    color: '#5d59cb',
    fontWeight: '300',
    letterSpacing: 2.5,
    fontFamily: 'open-sans-regular',
  },
  paginationDotStyle: {
    width: 10,
    height: 10,
    borderRadius: 5,
    marginHorizontal: 8,
    backgroundColor: 'rgba(93, 89, 203, 0.92)',
  },
});

export default styles;
