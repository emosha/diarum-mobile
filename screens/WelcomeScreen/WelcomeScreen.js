import React from 'react';
import PropTypes from 'prop-types';
import { View, Dimensions, ImageBackground } from 'react-native';
import Carousel, { Pagination } from 'react-native-snap-carousel';
import { Button } from 'react-native-elements';

import SlideImage from '../../components/SlideImage/SlideImage';
import styles from './styles';
import AppStyles from '../../App.styles';

const dimensions = Dimensions.get('window');
const imageWidth = dimensions.width;

export default class WelcomeScreen extends React.Component {
  static propTypes = {
    navigation: PropTypes.object.isRequired,
  }

  state = {
    activeSlide: 0,
    entries: [
      { imageSrc: require('../../assets/images/diarum-welcome-screen-one.png') },
      { imageSrc: require('../../assets/images/diarum-welcome-screen-two.png') },
      { imageSrc: require('../../assets/images/diarum-welcome-screen-three.png') },
    ],
  }

  /**
   * Renders each image slide component
   *
   * @param {object} image item object
   */
  renderSlideImage({ item }) {
    return <SlideImage imageSrc={item.imageSrc}/>;
  }

  get pagination() {
    const { entries, activeSlide } = this.state;
    return (
      <Pagination
        dotsLength={entries.length}
        activeDotIndex={activeSlide}
        containerStyle={styles.pagination}
        dotStyle={styles.paginationDotStyle}
        inactiveDotOpacity={0.4}
        inactiveDotScale={0.6}
      />
    );
  }

  /**
   * handles snap to item
   *
   * @param {number} index
   *
   * @returns {void}
   */
  handleSnapToItem = index => this.setState({ activeSlide: index });


  /**
   * navigates to signup page
   *
   * @returns {void}
   */
  navigateToSignup = () => this.props.navigation.navigate('Signup');

  render() {
    return (
      <ImageBackground source={require('../../assets/images/diarum-welcome.png')} style={styles.imageBackground}>
        <View style={styles.container}>
          <View style={AppStyles.authTop}>
            <Carousel
              data={this.state.entries}
              renderItem={this.renderSlideImage}
              onSnapToItem={this.handleSnapToItem}
              sliderWidth={imageWidth}
              itemWidth={imageWidth}
            />
            {this.pagination}
          </View>
          <View style={AppStyles.authBottom}>
            <Button
              raised
              title="GET STARTED"
              buttonStyle={AppStyles.buttonStyle}
              containerStyle={AppStyles.buttonContainerStyle}
              titleStyle={AppStyles.buttonTitleStyle}
              onPress={this.navigateToSignup}
            />
          </View>
        </View>
      </ImageBackground>
    );
  }
}
