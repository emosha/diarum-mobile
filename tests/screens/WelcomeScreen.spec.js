import React from 'react';
import WelcomeScreen from '../../screens/WelcomeScreen/WelcomeScreen';

jest.mock('react-native-snap-carousel');

test('renders correctly', () => {
  const tree = renderer.create(<WelcomeScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
