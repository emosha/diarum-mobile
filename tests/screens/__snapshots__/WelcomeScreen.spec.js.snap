// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen renders correctly 1`] = `
<WelcomeScreen
  navigation={
    Object {
      "navigate": [MockFunction],
    }
  }
>
  <ImageBackground
    source={1}
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <Component
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <Component
          source={1}
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": "100%",
                "width": "100%",
              },
              undefined,
            ]
          }
        >
          <Image
            source={1}
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "height": "100%",
                  "width": "100%",
                },
                undefined,
              ]
            }
          />
        </Component>
        <Component
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Component
              style={
                Object {
                  "backgroundColor": "transparent",
                  "flex": 1.88,
                  "position": "relative",
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "flex": 1.88,
                    "position": "relative",
                  }
                }
              >
                <Carousel
                  data={
                    Array [
                      Object {
                        "imageSrc": 1,
                      },
                      Object {
                        "imageSrc": 1,
                      },
                      Object {
                        "imageSrc": 1,
                      },
                    ]
                  }
                  itemWidth={750}
                  onSnapToItem={[Function]}
                  renderItem={[Function]}
                  sliderWidth={750}
                >
                  <Component>
                    <View>
                      <Component>
                        <Text>
                          Carousel
                        </Text>
                      </Component>
                    </View>
                  </Component>
                </Carousel>
                <Pagination
                  activeDotIndex={0}
                  containerStyle={
                    Object {
                      "backgroundColor": "transparent",
                      "bottom": "-5%",
                      "position": "absolute",
                      "width": "100%",
                    }
                  }
                  dotStyle={
                    Object {
                      "backgroundColor": "rgba(93, 89, 203, 0.92)",
                      "borderRadius": 5,
                      "height": 10,
                      "marginHorizontal": 8,
                      "width": 10,
                    }
                  }
                  dotsLength={3}
                  inactiveDotOpacity={0.4}
                  inactiveDotScale={0.6}
                >
                  <Component>
                    <View>
                      <Component>
                        <Text>
                          Pagination
                        </Text>
                      </Component>
                    </View>
                  </Component>
                </Pagination>
              </View>
            </Component>
            <Component
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <ForwardRef(Themed.Button)
                  buttonStyle={
                    Object {
                      "backgroundColor": "#F5BDBA",
                      "borderRadius": 0,
                      "padding": 15,
                      "paddingLeft": 40,
                      "paddingRight": 40,
                    }
                  }
                  containerStyle={
                    Object {
                      "marginTop": 35,
                    }
                  }
                  onPress={[Function]}
                  raised={true}
                  title="GET STARTED"
                  titleStyle={
                    Object {
                      "color": "#5d59cb",
                      "fontFamily": "open-sans-regular",
                      "fontSize": 25,
                      "fontWeight": "300",
                      "letterSpacing": 2.5,
                      "textAlign": "center",
                    }
                  }
                >
                  <ThemedComponent
                    buttonStyle={
                      Object {
                        "backgroundColor": "#F5BDBA",
                        "borderRadius": 0,
                        "padding": 15,
                        "paddingLeft": 40,
                        "paddingRight": 40,
                      }
                    }
                    containerStyle={
                      Object {
                        "marginTop": 35,
                      }
                    }
                    forwardedRef={null}
                    onPress={[Function]}
                    raised={true}
                    title="GET STARTED"
                    titleStyle={
                      Object {
                        "color": "#5d59cb",
                        "fontFamily": "open-sans-regular",
                        "fontSize": 25,
                        "fontWeight": "300",
                        "letterSpacing": 2.5,
                        "textAlign": "center",
                      }
                    }
                  >
                    <Button
                      TouchableComponent={[Function]}
                      buttonStyle={
                        Object {
                          "backgroundColor": "#F5BDBA",
                          "borderRadius": 0,
                          "padding": 15,
                          "paddingLeft": 40,
                          "paddingRight": 40,
                        }
                      }
                      containerStyle={
                        Object {
                          "marginTop": 35,
                        }
                      }
                      disabled={false}
                      iconRight={false}
                      loading={false}
                      onPress={[Function]}
                      raised={true}
                      theme={
                        Object {
                          "colors": Object {
                            "disabled": "hsl(208, 8%, 90%)",
                            "divider": "#bcbbc1",
                            "error": "#ff190c",
                            "grey0": "#393e42",
                            "grey1": "#43484d",
                            "grey2": "#5e6977",
                            "grey3": "#86939e",
                            "grey4": "#bdc6cf",
                            "grey5": "#e1e8ee",
                            "greyOutline": "#bbb",
                            "platform": Object {
                              "android": Object {
                                "error": "#f44336",
                                "primary": "#2196f3",
                                "secondary": "#9C27B0",
                                "success": "#4caf50",
                                "warning": "#ffeb3b",
                              },
                              "ios": Object {
                                "error": "#ff3b30",
                                "primary": "#007aff",
                                "secondary": "#5856d6",
                                "success": "#4cd964",
                                "warning": "#ffcc00",
                              },
                            },
                            "primary": "#2089dc",
                            "searchBg": "#303337",
                            "secondary": "#8F0CE8",
                            "success": "#52c41a",
                            "warning": "#faad14",
                          },
                        }
                      }
                      title="GET STARTED"
                      titleStyle={
                        Object {
                          "color": "#5d59cb",
                          "fontFamily": "open-sans-regular",
                          "fontSize": 25,
                          "fontWeight": "300",
                          "letterSpacing": 2.5,
                          "textAlign": "center",
                        }
                      }
                      type="solid"
                    >
                      <Component
                        style={
                          Object {
                            "backgroundColor": "#fff",
                            "borderRadius": 3,
                            "marginTop": 35,
                            "shadowColor": "rgba(0,0,0, .4)",
                            "shadowOffset": Object {
                              "height": 1,
                              "width": 1,
                            },
                            "shadowOpacity": 1,
                            "shadowRadius": 1,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "backgroundColor": "#fff",
                              "borderRadius": 3,
                              "marginTop": 35,
                              "shadowColor": "rgba(0,0,0, .4)",
                              "shadowOffset": Object {
                                "height": 1,
                                "width": 1,
                              },
                              "shadowOpacity": 1,
                              "shadowRadius": 1,
                            }
                          }
                        >
                          <TouchableOpacity
                            activeOpacity={0.3}
                            disabled={false}
                            onPress={[Function]}
                          >
                            <AnimatedComponent
                              accessible={true}
                              isTVSelectable={true}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Array [
                                  undefined,
                                  Object {
                                    "opacity": 1,
                                  },
                                ]
                              }
                            >
                              <Component
                                accessible={true}
                                isTVSelectable={true}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "opacity": 1,
                                  }
                                }
                              >
                                <View
                                  accessible={true}
                                  isTVSelectable={true}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    Object {
                                      "opacity": 1,
                                    }
                                  }
                                >
                                  <Component
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "backgroundColor": "#F5BDBA",
                                        "borderColor": "#2089dc",
                                        "borderRadius": 0,
                                        "borderWidth": 0,
                                        "flexDirection": "row",
                                        "justifyContent": "center",
                                        "padding": 15,
                                        "paddingLeft": 40,
                                        "paddingRight": 40,
                                      }
                                    }
                                  >
                                    <View
                                      style={
                                        Object {
                                          "alignItems": "center",
                                          "backgroundColor": "#F5BDBA",
                                          "borderColor": "#2089dc",
                                          "borderRadius": 0,
                                          "borderWidth": 0,
                                          "flexDirection": "row",
                                          "justifyContent": "center",
                                          "padding": 15,
                                          "paddingLeft": 40,
                                          "paddingRight": 40,
                                        }
                                      }
                                    >
                                      <Component
                                        style={
                                          Object {
                                            "color": "#5d59cb",
                                            "fontFamily": "open-sans-regular",
                                            "fontSize": 25,
                                            "fontWeight": "300",
                                            "letterSpacing": 2.5,
                                            "paddingBottom": 1,
                                            "paddingTop": 2,
                                            "textAlign": "center",
                                          }
                                        }
                                      >
                                        <Text
                                          style={
                                            Object {
                                              "color": "#5d59cb",
                                              "fontFamily": "open-sans-regular",
                                              "fontSize": 25,
                                              "fontWeight": "300",
                                              "letterSpacing": 2.5,
                                              "paddingBottom": 1,
                                              "paddingTop": 2,
                                              "textAlign": "center",
                                            }
                                          }
                                        >
                                          GET STARTED
                                        </Text>
                                      </Component>
                                    </View>
                                  </Component>
                                </View>
                              </Component>
                            </AnimatedComponent>
                          </TouchableOpacity>
                        </View>
                      </Component>
                    </Button>
                  </ThemedComponent>
                </ForwardRef(Themed.Button)>
              </View>
            </Component>
          </View>
        </Component>
      </View>
    </Component>
  </ImageBackground>
</WelcomeScreen>
`;
